### DISCUSSION TITLE
How to set Mocks for GraphQL queries ?

### DISCUSSION THREAD

**Discussion by kolchakn** [wiremock/WireMock.Net#1025] [Q&A] [answered]
Currently I have a lot of graphql queries which I need to mock. What is the best way to map a mock for each one.
- I see that `GraphQLMatcher` is suitable to verifying that a query/mutation is in the correct format.
- `JsonPathMatcher` checks the json body, but graphql POST body is not in json format.

I can't manage to make it work copy/pasting the body (especially hard when there are a lot of arguments).
What is the right way to do it (if it's possible at the moment) ?

**Comment by kolchakn** [comment]
I can't figure out how Strawberry Shake Client (which query I'm trying to mock) generates it's body, so i can match it exactly.

**Comment by StefH** [comment]
@kolchakn 
The body which is created by the Strawberry Shake Client should comply to the GraphQL schema which is exposed by the server.

So if you can get that GraphQL schema, you can use the GraphQLMatcher to match that schema.

**Comment by StefH** [comment, marked as answer]
@kolchakn 
Did this answer your question?

**Reply by kolchakn** [reply]
No, I ended up using
```
   ServiceProvider.WireMockServer.Given(Request.Create().WithPath("/graphql")
         .WithBody(new RegexMatcher($"^.*{queryRegexPattern}.*$")).UsingPost())
         .RespondWith(Response.Create().WithBody(response, encoding: Encoding.UTF8)
             .WithStatusCode(HttpStatusCode.OK));
```
where _queryRegexPattern_ is the name of the query
